erts/emulator/beam/erl_unicode.c

```diff
@@ -885,15 +885,29 @@ static BIF_RETTYPE characters_to_utf8_trap(BIF_ALIST_4)
 
 BIF_RETTYPE unicode_bin_is_7bit_1(BIF_ALIST_1)
 {
-    Sint need;
-    if(!is_bitstring(BIF_ARG_1)) {
+    const byte *temp_alloc = NULL, *bytes;
+    Uint size;
+    Eterm ret;
+
+    if (!is_bitstring(BIF_ARG_1)) {
```

Comment:
You can skip the is_bitstring test as erts_get_aligned_binary_bytes does that (again).